client/request_handler.go

```diff
@@ -2,32 +2,46 @@
 package client
 
 import (
+	"context"
 	"net/http"
 	"net/url"
 	"os"
 	"strings"
 )
 
 type RequestHandler struct {
-	Client BaseClient
-	Edge   string
-	Region string
+	Client            BaseClient
+	Edge              string
+	Region            string
+	clientWithContext BaseClientWithContext
 }
 
 func NewRequestHandler(client BaseClient) *RequestHandler {
+	// If the base client supports context, add it to the request handler.
+	// Otherwise we leave it nil and the base client will be used.
+	clientWithContext, _ := client.(BaseClientWithContext)
```

Comment:
Admittedly this is less than ideal, but it allows us to add Context support without breaking backwards compatibility.